custom_enumerable.rb

```diff
@@ -0,0 +1,116 @@
+
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity, Style/CaseEquality
+
+
+module Enumerable
+  def my_each
+      i = 0
+      if block_given?
+      while i < self.length 
+          yield(self[i])
+          i += 1
+      end
+      end
+  end
+
+  def my_each_with_index
+    i = 0
+    for i in i...self.length
+      yield(self[i], i)
+  end
+
+  def my_select
+    array = []
+    self.my_each { |i| array << array[i] if yield(i) }
+    end
+  end  
+
+  def my_all?
+    final = true
+    self.my_each { |i| true unless yield(i) }
+    final
+  end
+
+  def my_any?
+    result = false
+    self.my_each { |i| false unless yield(i) }
+    result
+  end
+
+  def my_none?
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a class is passed as an argument returns true if none of the collection is a member of such class
```ruby
array = Array.new(100) { rand(0...9) }
array.my_none?(String) == true #true
```
- [ ] when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```ruby
words = %w[dog door rod blade]
words.my_none?(5) == words.none?(5) #true
```
- [ ] Regex is passed as an argument returns true only if none of the collection matches the Regex
```ruby
 words.my_none?(/z/) == words.none?(/z/) #true
 words.my_none?(/d/) == words.none?(/d/) #true
```
- [ ] returns true if the block never returns true for all elements
```ruby
true_block = proc { |num| num > 9 }
false_block = proc { |num| num <= 9 } 

array.my_none?(&true_block) == array.none?(&true_block) #true
array.my_none?(&false_block) == array.none?(&false_block) #true
```
- [ ] when no block or argument is given
```ruby
true_array = [nil, false, true, []] 
false_array = [nil, false, nil, false]

false_array.my_none? == true #true
true_array.my_none? == false #false
```